### PR TITLE
motorway/contrib/amazone_sqs: raise exception when sqs message is to big

### PR DESCRIPTION
**Pull request by Maxtaager**

### CODE REVIEW

File: motorway/contrib/amazon_sqs/utils.py [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+SQS_MESSAGE_MAXIMUM_SIZE = 262144
 
 
 class SQSJSONMessage(Message):
     def encode(self, value):
-        return json.dumps(value)
+        s = json.dumps(value)
+        message_size = len(s.encode('utf-8'))
+        if message_size > SQS_MESSAGE_MAXIMUM_SIZE:
+            raise SQSMessageToLarge('message was %s in bytes and cant be bigger than %s' % (message_size, SQS_MESSAGE_MAXIMUM_SIZE))
```
Comment (alesdotio) [inline motorway/contrib/amazon_sqs/utils.py]: ``'Message size of %sB exceeds the maximum SQS message size of %sB'``

File: motorway/contrib/amazon_sqs/utils.py [hunk 2]
```diff
@@ -1,10 +1,17 @@
 import json
 from boto.sqs.message import Message
+from motorway.contrib.amazon_sqs.exceptions import SQSMessageToLarge
+
+SQS_MESSAGE_MAXIMUM_SIZE = 262144
```
Comment (alesdotio) [inline motorway/contrib/amazon_sqs/utils.py]: Add a link to the SQS documentation stating this.

Comment (Maxtaager) [reply motorway/contrib/amazon_sqs/utils.py]: fixed

File: motorway/contrib/amazon_sqs/utils.py (hunk 1 above)
Comment (Maxtaager) [reply motorway/contrib/amazon_sqs/utils.py]: fixed

**Comment by Maxtaager** [comment]
i have tested it with one byte over and with a json encoded string being exact 262144 bytes long. And there is no hidden overhead or anything, so it works as expected.

**Pull request merged**